prog/encodingexec.go

```diff
@@ -49,13 +79,27 @@ func (p *Prog) SerializeForExec(buffer []byte, pid int) error {
 	for _, c := range p.Calls {
 		// Calculate checksums.
 		csumMap := calcChecksumsCall(c, pid)
+		csumUses := make(map[*Arg]bool)
```

Comment:
Put `make(map[*Arg]bool)` under `if csumMap != nil`
This is on hot path and will generate garbage even when checksums are not used.